kernel/core/io/file.cpp

```diff
@@ -4241,10 +4293,14 @@ namespace NumeRe
                             {
                                 styleId = style->IntAttribute("numFmtId");
 
-                                // Those are time formats: 14-22, 45-47
-                                if ((styleId >= 14 && styleId <= 22) || (styleId >= 45 && styleId <= 47))
+                                // Check if styleId is in vDateTimeIds set
+                                if (vDateTimeIds.find(styleId) != vDateTimeIds.end())

```

Comment:
Instead of hardcoded checking, see if my set has any of styleID

Reply:
Your set probably won't contain the standard IDs as written in the comment:
`// Those are time formats: 14-22, 45-47`
I'd recommend to use them together with your solution like so

```C++
if ((styleId >= 14 && styleId <= 22) 
    || (styleId >= 45 && styleId <= 47) 
    || vDateTimeIds.find(styleId) != vDateTimeIds.end())
```